library/libc/print.c

```diff
@@ -6,21 +6,22 @@
  */
 
 #include "egos.h"
+#include "servers.h"
 #include <unistd.h>
 
 /* earth->vsprintf is the C library vsprintf, which converts
  * a format and arguments (e.g., ("example int=%d", 100)) into
  * a string. The string is passed to earth->tty_write() for printing.
  */
 
-#define LOG(x, y)  earth->tty_write(x, sizeof(x)); \
+#define LOG(x, y)  term_write(x, sizeof(x)); \
                    va_list args; \
                    va_start(args, format); \
                    char str_to_print[256]; \
                    uint len = earth->format_to_str(str_to_print, format, args); \
-                   earth->tty_write(str_to_print, len); \
+                   term_write(str_to_print, len); \
                    va_end(args); \
-                   earth->tty_write(y, sizeof(y));
+                   term_write(y, sizeof(y));
```

Comment:
Currently, a user app will send 3 messages to the terminal server for each `LOG`. Try to put everything in `str_to_print` and invoke `term_write` only once, i.e., sending a single message.

Reply:
Currently solve the issue using `strcpy` and `strcat`, but let me know if you believe there are cleaner ways of constructing a single string in `LOG`